File: src/main/java/org/example/kalkulationsprogramm/domain/Produktkategorie.java
package org.example.kalkulationsprogramm.domain;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.util.ArrayList;
import java.util.List;

@Getter
@Setter

@Entity
public class Produktkategorie
{
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id; // Original: ProduktkategorieID

    @Column(nullable = false)
    private String bezeichnung;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private Verrechnungseinheit verrechnungseinheit;

    private String bildUrl;

    @Column(columnDefinition = "TEXT")
    private String beschreibung;


    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "parent_kategorie_id") // Original: Unterkategorie
    private Produktkategorie uebergeordneteKategorie;

    @OneToMany(mappedBy = "uebergeordneteKategorie")
    private List<Produktkategorie> unterkategorien = new ArrayList<>();

    @OneToMany(mappedBy = "produktkategorie")
    private List<ProjektProduktkategorie> projektProduktkategorien = new ArrayList<>();

}

File: src/main/kotlin/org/example/kalkulationsprogramm/domain/Produktkategorie.kt
package org.example.kalkulationsprogramm.domain

import jakarta.persistence.*

@Entity
open class Produktkategorie {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    open var id: Long? = null

    @Column(nullable = false)
    open var bezeichnung: String? = null

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    open var verrechnungseinheit: Verrechnungseinheit? = null

    open var bildUrl: String? = null

    @Column(columnDefinition = "TEXT")
    open var beschreibung: String? = null

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "parent_kategorie_id") // Original: Unterkategorie
    open var uebergeordneteKategorie: Produktkategorie? = null

    @OneToMany(mappedBy = "uebergeordneteKategorie")
    open var unterkategorien: MutableList<Produktkategorie> = ArrayList()

    @OneToMany(mappedBy = "produktkategorie")
    open var projektProduktkategorien: MutableList<ProjektProduktkategorie> = ArrayList()

}